src/world.cpp

```diff
@@ -102,8 +113,15 @@ World::World(
         macro_tile_size, height, 5, tile_type, materials, biome_data["Biome_1"],
         get_grass_grad_data(materials_json),
         materials_json["Dirt"]["Gradient"]["midpoint"].asInt()
-    ) {}
+    ) {
+    // on initialization world reserves the space it would need for shared pointers
+    chunks_mesh.resize(terrain_main.get_chunks().size());
+    for (size_t i = 0; i < chunks_mesh.size(); i++) {
+        chunks_mesh[i] = std::make_shared<terrain::TerrainMesh>();
+    }
```

Comment:
🚨Code duplication🚨

Reply:
it's three lines of code

Reply:
and if you change it in one place you have to remember to change it in three.